--- parse.c
+++ parse.c
@@ -267,10 +267,10 @@ int isarg(const char *s)		/* is s in argument list for current function? */
 
 int ptoi(void *p)	/* convert pointer to integer */
 {
-	return (int) (long) p;	/* swearing that p fits, of course */
+	return (int) (intptr_t) p;	/* swearing that p fits, of course */
 }
 
 Node *itonp(int i)	/* and vice versa */
 {
-	return (Node *) (long) i;
+	return (Node *) (intptr_t) i;
 }
